From b02b24b3b9b99be7a8050181ecd8e347cf6dea4c Mon Sep 17 00:00:00 2001
From: Andreas Ternstedt <a.ternstedt@setek.se>
Date: Tue, 4 Nov 2025 15:17:32 +0100
Subject: [PATCH 5/5] imx8mp_var_dart: secure autoboot by disabling keyboard interrupt

Set CONFIG_AUTOBOOT_KEYED=y with empty stop string to prevent
unauthorized U-Boot console access via keyboard interrupt.

Co-developed-by: Albin Söderqvist <albin@albinutils.se>
Signed-off-by: Andreas Ternstedt <a.ternstedt@setek.se>

---
 configs/imx8mp_var_dart_defconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configs/imx8mp_var_dart_defconfig b/configs/imx8mp_var_dart_defconfig
index a3d7f568a91..14df37414db 100644
--- a/configs/imx8mp_var_dart_defconfig
+++ b/configs/imx8mp_var_dart_defconfig
@@ -13,6 +13,8 @@ CONFIG_SYS_I2C_MXC_I2C2=y
 CONFIG_SYS_I2C_MXC_I2C3=y
 CONFIG_SYS_I2C_MXC_I2C4=y
 CONFIG_DM_GPIO=y
+CONFIG_AUTOBOOT_KEYED=y
+CONFIG_AUTOBOOT_STOP_STR=""
 CONFIG_DEFAULT_DEVICE_TREE="imx8mp-var-dart-dt8mcustomboard"
 CONFIG_SPL_TEXT_BASE=0x920000
 CONFIG_TARGET_IMX8MP_VAR_DART=y
-- 
2.34.1

